### ISSUE TITLE
Broken notifications with no error messages

### ISSUE THREAD

**Issue by GraemeMeyerGT** [LuatixHQ/opencti#3460]
## Description

We have two OpenCTI instances with identical configs, and notifications seem to be completely broken in both. No notifications are ever triggered or presented, despite a variety of configurations tested for triggers and digests tested. The "Notification generation manager" and "Notification publishing manager" tools/services are both showing as enabled.

## Environment

1. OS (where OpenCTI server runs): { e.g. Mac OS 10, Windows 10, Ubuntu 16.4, etc. } Docker 24.0.1 and Docker Compose version v2.18.1 on Ubuntu 22 (we use the provided Docker Compose configuration)
2. OpenCTI version: { e.g. OpenCTI 1.0.2 } 5.7.6
3. OpenCTI client: { e.g. frontend or python } N/A

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Create a trigger for any create/update/delete of any entity (I tested reports and indicators). 
2. Perform the event that should trigger the notification (e.g. create a report)
3. Nothing happens
4. Nothing in the UI
5. Nothing in the logs

## Expected Output

I expect notifications to be created in the UI on the `/dashboard/profile/notifications` page

## Actual Output

Nothing in the UI, nothing in the logs to indicate any awareness of notification activity, let alone errors or debug information.

## Additional information

I realised this wasn't working while I was trying to configure SMTP in the platform, so I have now tried both SMTP and UI notifications, and neither has any impact or shows any events.

## Screenshots (optional)
![image](https://github.com/OpenCTI-Platform/opencti/assets/107847353/6e020ae4-f842-4bba-acdd-f45cdaccaa97)
![image](https://github.com/OpenCTI-Platform/opencti/assets/107847353/2adabb74-34bc-4fb7-aa47-68609781154a)

**Comment by SarahBocognano** [comment]
Hi @GraemeMeyerGT , I'm sorry but it seems that it's working on my plateform and on the demo.

**Comment by GraemeMeyerGT** [comment]
Yes definitely could do with some debug tracing or something - is there a debug setting I can turn on to gather more info?

**Comment by GraemeMeyerGT** [comment]
I realised I can turn the `APP__APP_LOGS__LOGS_LEVEL` setting to debug, so I've done that, tried again to create a live trigger for report creation or modification, and following are all the log events mentioning "trigger" or "notif" (case-insensitive grep):

```
opencti-opencti-1                                         | {"category":"APP","level":"warn","message":"[CHECK] SMTP seems down, email notification will may not work","timestamp":"2023-06-02T15:05:59.703Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"warn","message":"[CHECK] SMTP seems down, email notification will may not work","timestamp":"2023-06-02T15:06:02.585Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"debug","message":"[SEARCH] paginate","query":{"_source":true,"body":{"query":{"bool":{"must":[{"bool":{"minimum_should_match":1,"should":[{"match_phrase":{"entity_type.keyword":"Trigger"}},{"match_phrase":{"parent_types.keyword":"Trigger"}}]}}],"must_not":[]}},"size":5000,"sort":[{"standard_id.keyword":"asc"}]},"ignore_throttled":false,"index":["opencti_internal_objects*","opencti_stix_meta_objects*","opencti_stix_domain_objects*","opencti_stix_cyber_observables*","opencti_inferred_entities*"],"track_total_hits":true},"timestamp":"2023-06-02T15:06:17.631Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"debug","message":"[SEARCH] paginate","query":{"_source":true,"body":{"query":{"bool":{"must":[{"bool":{"minimum_should_match":1,"should":[{"match_phrase":{"entity_type.keyword":"Trigger"}},{"match_phrase":{"parent_types.keyword":"Trigger"}}]}}],"must_not":[]}},"size":5000,"sort":[{"standard_id.keyword":"asc"}]},"ignore_throttled":false,"index":["opencti_internal_objects*","opencti_stix_meta_objects*","opencti_stix_domain_objects*","opencti_stix_cyber_observables*","opencti_inferred_entities*"],"track_total_hits":true},"timestamp":"2023-06-02T15:06:19.310Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"debug","message":"[SEARCH] elCount","query":{"body":{"query":{"bool":{"must":[{"bool":{"minimum_should_match":1,"should":[{"match_phrase":{"entity_type.keyword":"Notification"}},{"match_phrase":{"parent_types.keyword":"Notification"}}]}},{"bool":{"minimum_should_match":1,"should":[{"multi_match":{"fields":["user_id.keyword"],"query":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"}}]}},{"bool":{"minimum_should_match":1,"should":[{"multi_match":{"fields":["is_read"],"query":"false"}}]}}],"must_not":[]}}},"index":"opencti_internal_objects*"},"timestamp":"2023-06-02T15:07:01.086Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"debug","message":"[SEARCH] paginate","query":{"_source":true,"body":{"query":{"bool":{"must":[{"bool":{"minimum_should_match":1,"should":[{"match_phrase":{"entity_type.keyword":"Notification"}},{"match_phrase":{"parent_types.keyword":"Notification"}}]}},{"bool":{"minimum_should_match":1,"should":[{"multi_match":{"fields":["user_id.keyword"],"query":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"}}]}},{"exists":{"field":"created"}}],"must_not":[]}},"size":25,"sort":[{"created":"desc"},{"standard_id.keyword":"asc"}]},"ignore_throttled":false,"index":["opencti_internal_objects*","opencti_stix_meta_objects*","opencti_stix_domain_objects*","opencti_stix_cyber_observables*","opencti_inferred_entities*"],"track_total_hits":true},"timestamp":"2023-06-02T15:07:03.259Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"debug","message":"[SEARCH] paginate","query":{"_source":true,"body":{"query":{"bool":{"must":[{"bool":{"minimum_should_match":1,"should":[{"match_phrase":{"entity_type.keyword":"Trigger"}},{"match_phrase":{"parent_types.keyword":"Trigger"}}]}},{"bool":{"minimum_should_match":1,"should":[{"multi_match":{"fields":["user_ids.keyword"],"query":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"}}]}},{"exists":{"field":"name.keyword"}}],"must_not":[]}},"size":25,"sort":[{"name.keyword":"asc"},{"standard_id.keyword":"asc"}]},"ignore_throttled":false,"index":["opencti_internal_objects*","opencti_stix_meta_objects*","opencti_stix_domain_objects*","opencti_stix_cyber_observables*","opencti_inferred_entities*"],"track_total_hits":true},"timestamp":"2023-06-02T15:07:04.754Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"debug","message":"[SEARCH] elInternalLoadById","query":{"_source":true,"body":{"query":{"bool":{"must":[{"bool":{"minimum_should_match":1,"should":[{"term":{"internal_id.keyword":"0fde6fc8-d77b-449e-bdc3-4acba1d8c005"}},{"term":{"standard_id.keyword":"0fde6fc8-d77b-449e-bdc3-4acba1d8c005"}},{"term":{"x_opencti_stix_ids.keyword":"0fde6fc8-d77b-449e-bdc3-4acba1d8c005"}}]}},{"bool":{"minimum_should_match":1,"should":[{"match_phrase":{"entity_type.keyword":"Trigger"}},{"match_phrase":{"parent_types.keyword":"Trigger"}}]}}],"must_not":[]}}},"ignore_throttled":false,"index":["opencti_internal_objects*","opencti_internal_relationships*","opencti_stix_meta_objects*","opencti_stix_meta_relationships*","opencti_stix_cyber_observable_relationships*","opencti_stix_domain_objects*","opencti_stix_core_relationships*","opencti_stix_sighting_relationships*","opencti_stix_cyber_observables*","opencti_inferred_entities*","opencti_inferred_relationships*"],"size":5000},"timestamp":"2023-06-02T15:07:06.184Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"debug","message":"[SEARCH] paginate","query":{"_source":true,"body":{"query":{"bool":{"must":[{"bool":{"minimum_should_match":1,"should":[{"multi_match":{"fields":["event_source_id.keyword"],"query":"import/Trigger/0fde6fc8-d77b-449e-bdc3-4acba1d8c005/"}}]}},{"bool":{"minimum_should_match":1,"should":[{"multi_match":{"fields":["event_type.keyword"],"query":"INTERNAL_EXPORT_FILE"}}]}},{"exists":{"field":"timestamp"}}],"must_not":[]}},"size":10,"sort":[{"timestamp":"desc"},{"standard_id.keyword":"asc"}]},"ignore_throttled":false,"index":"opencti_history*","track_total_hits":true},"timestamp":"2023-06-02T15:07:08.082Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"debug","message":"[SEARCH] paginate","query":{"_source":true,"body":{"query":{"bool":{"must":[{"bool":{"minimum_should_match":1,"should":[{"multi_match":{"fields":["event_source_id.keyword"],"query":"export/Trigger/0fde6fc8-d77b-449e-bdc3-4acba1d8c005/"}}]}},{"bool":{"minimum_should_match":1,"should":[{"multi_match":{"fields":["event_type.keyword"],"query":"INTERNAL_EXPORT_FILE"}}]}},{"exists":{"field":"timestamp"}}],"must_not":[]}},"size":10,"sort":[{"timestamp":"desc"},{"standard_id.keyword":"asc"}]},"ignore_throttled":false,"index":"opencti_history*","track_total_hits":true},"timestamp":"2023-06-02T15:07:08.083Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"debug","message":"[SEARCH] elInternalLoadById","query":{"_source":true,"body":{"query":{"bool":{"must":[{"bool":{"minimum_should_match":1,"should":[{"term":{"internal_id.keyword":"trigger--e356b0e1-4094-5e4c-931e-98761f797c18"}},{"term":{"standard_id.keyword":"trigger--e356b0e1-4094-5e4c-931e-98761f797c18"}},{"term":{"x_opencti_stix_ids.keyword":"trigger--e356b0e1-4094-5e4c-931e-98761f797c18"}}]}},{"bool":{"minimum_should_match":1,"should":[{"match_phrase":{"entity_type.keyword":"Trigger"}},{"match_phrase":{"parent_types.keyword":"Trigger"}}]}}],"must_not":[]}}},"ignore_throttled":false,"index":["opencti_internal_objects*","opencti_internal_relationships*","opencti_stix_meta_objects*","opencti_stix_meta_relationships*","opencti_stix_cyber_observable_relationships*","opencti_stix_domain_objects*","opencti_stix_core_relationships*","opencti_stix_sighting_relationships*","opencti_stix_cyber_observables*","opencti_inferred_entities*","opencti_inferred_relationships*"],"size":5000},"timestamp":"2023-06-02T15:07:37.043Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"debug","message":"[SEARCH] paginate","query":{"_source":true,"body":{"query":{"bool":{"must":[{"bool":{"minimum_should_match":1,"should":[{"match_phrase":{"entity_type.keyword":"Notification"}},{"match_phrase":{"parent_types.keyword":"Notification"}}]}},{"bool":{"minimum_should_match":1,"should":[{"multi_match":{"fields":["user_id.keyword"],"query":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"}}]}},{"exists":{"field":"created"}}],"must_not":[]}},"size":25,"sort":[{"created":"desc"},{"standard_id.keyword":"asc"}]},"ignore_throttled":false,"index":["opencti_internal_objects*","opencti_stix_meta_objects*","opencti_stix_domain_objects*","opencti_stix_cyber_observables*","opencti_inferred_entities*"],"track_total_hits":true},"timestamp":"2023-06-02T15:09:30.102Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"debug","message":"[SEARCH] paginate","query":{"_source":true,"body":{"query":{"bool":{"must":[{"bool":{"minimum_should_match":1,"should":[{"match_phrase":{"entity_type.keyword":"Trigger"}},{"match_phrase":{"parent_types.keyword":"Trigger"}}]}},{"bool":{"minimum_should_match":1,"should":[{"multi_match":{"fields":["user_ids.keyword"],"query":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"}}]}},{"exists":{"field":"name.keyword"}}],"must_not":[]}},"size":25,"sort":[{"name.keyword":"asc"},{"standard_id.keyword":"asc"}]},"ignore_throttled":false,"index":["opencti_internal_objects*","opencti_stix_meta_objects*","opencti_stix_domain_objects*","opencti_stix_cyber_observables*","opencti_inferred_entities*"],"track_total_hits":true},"timestamp":"2023-06-02T15:09:31.100Z","version":"5.7.6"}
opencti-opencti-1                                         | {"category":"APP","level":"debug","message":"[SEARCH] paginate","query":{"_source":true,"body":{"query":{"bool":{"must":[{"bool":{"minimum_should_match":1,"should":[{"match_phrase":{"entity_type.keyword":"Notification"}},{"match_phrase":{"parent_types.keyword":"Notification"}}]}},{"bool":{"minimum_should_match":1,"should":[{"multi_match":{"fields":["user_id.keyword"],"query":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"}}]}},{"exists":{"field":"created"}}],"must_not":[]}},"size":25,"sort":[{"created":"desc"},{"standard_id.keyword":"asc"}]},"ignore_throttled":false,"index":["opencti_internal_objects*","opencti_stix_meta_objects*","opencti_stix_domain_objects*","opencti_stix_cyber_observables*","opencti_inferred_entities*"],"track_total_hits":true},"timestamp":"2023-06-02T15:09:32.968Z","version":"5.7.6"}
```

It's hard for me to say, but doesn't look like there's much there. Maybe this area needs some more detailed debug logging?

**Comment by marieflorescontact** [comment]
It seems that we have a `smtp` issue. we are working on it. 
however UI notification is working properly on demo.

**Comment by GraemeMeyerGT** [comment]
@marieflorescontact  what is the smtp issue? I will be working on smtp in near future so would be good to know.

I have partially diagnosed and fixed the broken notifications issue. It was permissions/marking/groups/roles problem.

Couple of issues I identified due to our upgrade from 5.6.x to 5.7.x:

* New "Default" role wasn't given to people that had a group specified already (e.g. CTI Analysts)
* Existing custom groups were not given "Default" role
* Marking definitions setting for custom groups had been lost as well, so I've had to re-apply those manually
* "Automatically authorize this group to new marking definition" setting was not transitioned from Roles to newly created Groups during the migration
* Platform restart was required to apply organisation setting to the local Administrator user

After fixing all of the above, the notifications are now working in the web UI as expected.

tl;dr, this was likely all problems with the migration from 5.6.x to 5.7.x, and I wouldn't expect a clean 5.7 platform to experience any of them
